docs/src/guide/index.md

```diff
@@ -254,7 +254,7 @@ const knex = require('knex')({
 });
 ```
 
-By default, the configuration object received via a function is cached and reused for all connections. To change this behavior, an `expirationChecker` function can be returned as part of the configuration object. The `expirationChecker` is consulted before trying to create new connections, and in case it returns `true`, a new configuration object is retrieved. For example, to work with an authentication token that has a limited lifespan:
+By default, the configuration object received via a function is cached and reused for all connections. To change this behavior, an `expirationChecker` function can be returned as part of the configuration object. The `expirationChecker` is consulted both when creating new connections and before reusing a pooled connection. If it returns `true`, Knex refreshes the connection settings from your provider and drops the current connection so the pool can create a new one with the updated configuration. If the checker still reports expired after a refresh, it is disabled to avoid repeated churn.
```

Comment:
Docs should be updated to reflect the new behavior :)

Reply:
Ah I had missed this one. Done.